packages/atomic/src/components/search/result-template-components/atomic-result-image/atomic-result-image.tsx

```diff
@@ -36,27 +41,57 @@ export class AtomicResultImage implements InitializableComponent {
     return Array.isArray(value) ? value[0] : value;
   }
 
-  public render() {
-    const url = this.url;
+  private logWarning(message: string) {
+    this.bindings.engine.logger.warn(message, this.host);
+  }
 
-    if (!url) {
-      this.host.remove();
+  private handleImageError(target: EventTarget | null) {
+    const image = target as HTMLImageElement;
+
+    if (image.src === this.fallback) {
       return;
     }
 
-    if (typeof url !== 'string') {
-      this.bindings.engine.logger.error(
-        `Expected "${this.field}" to be a text field.`,
-        this.host
-      );
+    if (this.fallback) {
+      image.src = this.fallback;
```

Comment:
Hadn't thought of it until now, but we should avoid modifying attributes directly, since they'll be reset whenever there's a re-render of the component.

Instead of dynamically modifying `image.src`, I would recommend adding a `@State()` property, perhaps something like `@State() useFallback = false`. Then, in the `render` function, you'd just have to check whether `useFallback` is true to determine which URL to use.

Reply:
I did not know changing a state would recall the render function. Thanks